Source/Extensions/Blazorise.LottieAnimation/wwwroot/lottie-animation.js

```diff
@@ -0,0 +1,80 @@
+import './vendors/lottie.js?v=1.1.2.0';
+
+import { getRequiredElement } from "../Blazorise/utilities.js?v=1.1.2.0";
+
+/**
+ * Initializes a new animation instance
+ * @param dotNetAdapter Reference to the dotnet object
+ * @param element Container element reference
+ * @param elementId Container element Id
+ * @param options Animation configuration options
+ * @returns Reference to the Lottie Animation
+ */
+export function initializeAnimation(dotNetAdapter, element, elementId, options) {
+    element = getRequiredElement(element, elementId);
+
+    if (!element) {
+        return;
+    }
+
+    options.container = element;
+
+    const animation = lottie.loadAnimation(options);
+    animation.setDirection(options.direction);
+    animation.setSpeed(options.speed);
+
+    // Add a function to set a flag for whether or not we're sending current frame updates
+    animation.setSendCurrentFrame = function (shouldSend) {
+        this.sendCurrentFrame = shouldSend;
+    }
+
+    // Add a function to set the loop property
+    animation.setLoop = function (loop) {
+        this.loop = loop;
+    }
+
+    animation.setSendCurrentFrame(options.sendCurrentFrame);
+    registerEvents(dotNetAdapter, animation);
+
+    return animation;
+}
+
+async function invokeDotNetMethodAsync(dotNetAdapter, methodName, ...args) {
```

Comment:
We're trying to target the ES6 compatibility level. Can you make sure async/await is OK here?

Reply:
Looks like async/await wasn't added until ECMAScript 2017, so I've swapped it out for the Promise APIs.

Reply:
Looking good now, thanks.